pages/api/enroll/child.ts

```diff
@@ -17,6 +19,22 @@ export default async function handle(
     req: NextApiRequest,
     res: NextApiResponse,
 ): Promise<void> {
+    const session = await getSession({ req });
+
+    // If there is no session or the user is not a parent
+    if (!session || session.role !== roles.PARENT) {
+        return ResponseUtil.returnUnauthorized(res, "Unauthorized");
+    }
+
+    const parentId = session.id as number;
+    if (!parentId) {
+        // TODO: What should the error message be here?
```

Comment:
what you have LGTM - we control what's stored in the session so it should likely never get to this point anyway